src/commonMain/kotlin/exchange.dydx.abacus/state/manager/notification/NotificationsProvider.kt

```diff
@@ -0,0 +1,74 @@
+package exchange.dydx.abacus.state.manager.notification
+
+import exchange.dydx.abacus.output.Asset
+import exchange.dydx.abacus.output.Notification
+import exchange.dydx.abacus.protocols.ParserProtocol
+import exchange.dydx.abacus.state.manager.V4Environment
+import exchange.dydx.abacus.state.manager.notification.providers.BlockRewardNotificationProvider
+import exchange.dydx.abacus.state.manager.notification.providers.OrderStatusChangesNotificationProvider
+import exchange.dydx.abacus.state.model.TradingStateMachine
+import exchange.dydx.abacus.utils.IMap
+import exchange.dydx.abacus.utils.JsonEncoder
+import exchange.dydx.abacus.utils.ParsingHelper
+import exchange.dydx.abacus.utils.UIImplementations
+import kollections.toIMap
+
+interface NotificationsProviderProtocol {
+    fun buildNotifications(
+        stateMachine: TradingStateMachine,
+        subaccountNumber: Int
+    ): IMap<String, Notification>
```

Comment:
I think I'm missing some context: why are notifications modeled as a Map and not a List?

Reply:
The map is used for post-processing of the messages. See consolidateNotifications()